src/libcore/cell.rs

```diff
@@ -369,6 +367,7 @@ impl<T> Cell<T> {
         if ptr::eq(self, other) {
             return;
         }
+        // SAFETY: not threadsafe, but it's OK since we know `Cell` isn't threadsafe
```

Comment:
Besides data races, a major concern here is invalidating pointers. The reason this is safe is that `Cell` rules out interior pointers -- there can be nothing pointing *into* either of these `Cell`, so we can swap out their content just fine.

This applies to the other unsafe blocks in this file as well.

Reply:
Also, "not threadsafe is okay" is negative reasoning, that's somewhat backwards. What we need here is *positive* evidence that there cannot be a race, and that evidence is that `Cell` is `!Sync`. I think that's what you mean by your second use of the word "threadsafe" but it's not entirely clear.